opentelemetry-sdk/src/opentelemetry/sdk/trace/export/experimental/timer.py

```diff
@@ -0,0 +1,111 @@
+import abc
+import threading
+import typing
+
+
+class TimerABC(abc.ABC):
+    """
+    An interface extracted from PeriodicTimer so alternative implementations can be used for testing.
+    """
+
+    @abc.abstractmethod
+    def set_callback(self, cb) -> None:
+        pass
+
+    @abc.abstractmethod
+    def start(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def poke(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def stop(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def started(self) -> None:
+        pass
+
+    @abc.abstractmethod
+    def stopped(self) -> None:
+        pass
+
+
+class PeriodicTimer(TimerABC):
+    """
+    Executes the passed-in callback on a timer at the specified interval. The callback can be run sooner than the
+    interval via the poke() method, which also resets the timer.
+    """
+
+    def __init__(
+        self,
+        interval_sec: int,
+        callback: typing.Callable[[], None] = lambda: None,
+        daemon: bool = True,
+    ):
+        self._callback = callback
+        self._interval_sec = interval_sec
+        self._stop = threading.Event()
+        self._poke = threading.Event()
+        self._thread = threading.Thread(target=self._work, daemon=daemon)
+
+    def set_callback(self, callback: typing.Callable[[], None]) -> None:
+        self._callback = callback
+
+    def start(self) -> None:
+        self._thread.start()
+
+    def _work(self) -> None:
+        while True:
+            self._poke.wait(timeout=self._interval_sec)
+            if self._stop.is_set():
```

Comment:
If there is a stop event, it will not be received until we are done waiting for a poke, at most `self._interval_sec`. I'm guessing this is why we used a queue for this before.

Reply:
True. The `stop()` method [calls poke](https://github.com/open-telemetry/opentelemetry-python/pull/3465/files#diff-a2b9f69e2880dc0988069fac93565195fdf33a10c6f53f21f5fa05cda3afba25R77) for this reason. Does this address your concern?

Reply:
Resolving this conversation for now as I've added [a timer implementation](https://github.com/open-telemetry/opentelemetry-python/pull/3465/files#diff-a2b9f69e2880dc0988069fac93565195fdf33a10c6f53f21f5fa05cda3afba25R31) that's perhaps a little more straightforward.